### ISSUE TITLE
[BUG] Charset issue?

### ISSUE THREAD

**Issue by alessandrococco** [enricocid/Music-Player-GO#416]
Hello!

I noticed a weird issue with the title of a track:

![image](https://user-images.githubusercontent.com/2331798/169156965-7f12dc7f-0751-41c7-abfc-8bbde5584768.png)

Note: the separator is `–`, not `-`.

This is the same track but in [Omnia Player](https://play.google.com/store/apps/details?id=com.rhmsoft.omnia):
 
![image](https://user-images.githubusercontent.com/2331798/169157233-475c1d35-0d68-462e-8d95-836add1949b2.png)
 
![image](https://user-images.githubusercontent.com/2331798/169157295-86a6fb38-1976-4e83-baf2-d390de52cc9c.png)

**Smartphone (please complete the following information):**
 - Device: POCO X3 NFC
 - OS: Android 11, MIUI 12.5.7 Global

This is the file: [0103-of-matter-resist - Copy.zip](https://github.com/enricocid/Music-Player-GO/files/8720972/0103-of-matter-resist.-.Copy.zip)

**Comment by enricocid** [comment]
Thanks. Gonna try to fix tomorrow 🙏🏻😁

**Comment by enricocid** [comment]
Could you send me the original file?
Can't reproduce with this :)
![Uploading Screenshot_20220519-111025_MPGO_debug.png…]()

**Comment by alessandrococco** [comment]
> Could you send me the original file? Can't reproduce with this :)

Yeah sorry, I just noticed I messed up with the file before uploading it 😄 

I uploaded the tracks [here](https://ln5.sync.com/dl/ff8c02f20/a32869qf-gm8hv725-f6d79gyg-n9772ds9) (the link expires at 23:00).

**Comment by enricocid** [comment]
Kein problem 😁
I can now reproduce the issue. Thanks.

**Comment by enricocid** [comment]
@OxygenCobalt this issue should affect your player too ☺️ 
Tried various approaches but still can't find how to fix it.
Let me know if you find a solution 😁

**Comment by OxygenCobalt** [comment]
Hey @enricocid, I've discovered the cause of the issue in the android source. It's [this file, CharacterEncodingDetector](https://cs.android.com/android/platform/superproject/+/master:frameworks/av/media/libmedia/CharacterEncodingDetector.cpp). It's encoding logic is bugged and does not handle certain forms of UTF-8. In the next version of my app I'll be moving on from MediaStore as anything but a redundancy layer, but it might be helpful for you to file a bug report.
